src/metrics/hpr_metrics.erl

```diff
@@ -34,12 +35,35 @@
 start_link(Args) ->
     gen_server:start_link({local, ?SERVER}, ?SERVER, Args, []).
 
+-spec observe_packet_up(
+    PacketType :: hpr_packet_up:type(),
+    Status :: hpr_routing:hpr_routing_response(),
+    NumberOfRoutes :: non_neg_integer(),
+    Start :: non_neg_integer()
+) -> ok.
+observe_packet_up({Type, _}, RoutingStatus, NumberOfRoutes, Start) ->
+    Status =
+        case RoutingStatus of
+            ok -> ok;
+            {error, _} -> error
+        end,
+    prometheus_histogram:observe(
+        ?METRICS_PACKET_UP_HISTOGRAM,
+        [Type, Status, NumberOfRoutes],
+        erlang:system_time(millisecond) - Start
+    ).
 %% ------------------------------------------------------------------
 %% gen_server Function Definitions
 %% ------------------------------------------------------------------
 init(_Args) ->
     erlang:process_flag(trap_exit, true),
     lager:info("init"),
+    ElliOpts = [
+        {callback, hpr_metrics_handler},
+        {callback_args, #{}},
+        {port, 3000}
+    ],
+    {ok, _Pid} = elli:start_link(ElliOpts),
```

Comment:
Should this be started by the supervisor? It's somewhat independent of the this gen_server, isn't it?

Reply:
We do this in Packet Purchaser in the top level sup for reference.
https://github.com/helium/packet-purchaser/blob/master/src/pp_sup.erl#L93-L100

Reply:
👍🏻